behind_ingensec.rb

```diff
@@ -0,0 +1,285 @@
+        'data'   => "url=#{hostname}&submit=Check+Reverse+Ip+Domains"
+      })
+      response = cli.send_recv(request)
+      cli.close
+
+    rescue ::Rex::ConnectionError, Errno::ECONNREFUSED, Errno::ETIMEDOUT
+      print_error('HTTP connection failed to PrePost SEO website.')
+      return false
+    end
+
+    html  = response.get_html_document
+    table = html.css('table.table').first
+    rows  = table.css('tr')
+
+    arIps = []
+    rows.each_with_index.map do | row, index |
+      row = /(\d*\.\d*\.\d*\.\d*)/.match(row.css('td').map(&:text).to_s)
+      unless row.nil?
+        arIps.push(row)
+      end
+    end
+
+    if arIps.empty?
+      print_bad('No domain IP(s) history founds.')
+      return false
+    end
+
+    return arIps
+  end
+
+  ## auxiliary/gather/enum_dns.rb
+  def do_dns_enumeration(domain, threads)
+    wordlist = datastore['WORDLIST']
+    return if wordlist.blank?
+    threads  = 1 if threads <= 0
+
+    queue    = []
+    File.foreach(wordlist) do | line |
+      queue << "#{line.chomp}.#{domain}"
+    end
+
+    arIps = []
+    until queue.empty?
+      t = []
+      threads = 1 if threads <= 0
+
+      if queue.length < threads
+        # work around issue where threads not created as the queue isn't large enough
+        threads = queue.length
+      end
+
+      begin
+        1.upto(threads) do
+          t << framework.threads.spawn("Module(#{refname})", false, queue.shift) do | test_current |
+            Thread.current.kill unless test_current
+            a = /(\d*\.\d*\.\d*\.\d*)/.match(do_dns_get_a(test_current, 'DNS bruteforce records').to_s)
+            arIps.push(a) if a
+          end
+        end
+        t.map(&:join)
+
+      rescue ::Timeout::Error
+      ensure
+        t.each { | x | x.kill rescue nil }
+      end
+    end
+
+    if arIps.empty?
+      print_bad('No domain IP(s) history founds.')
+      return false
+    end
+
+    return arIps
+  end
+
+  ## auxiliary/gather/enum_dns.rb
+  def do_dns_get_a(domain, type='DNS A records')
+    response = do_dns_query(domain, 'A')
+    return if response.blank? || response.answer.blank?
+
+    response.answer.each do | row |
+      next unless row.class == Net::DNS::RR::A
+    end
+  end
+
+  ## auxiliary/gather/enum_dns.rb
+  def do_dns_query(domain, type)
+    begin
+      dns                = Net::DNS::Resolver.new
+      dns.use_tcp        = false
+      dns.udp_timeout    = 8
+      dns.retry_number   = 2
+      dns.retry_interval = 2
+      dns.query(domain, type)
+    rescue ResolverArgumentError, Errno::ETIMEDOUT, ::NoResponseError, ::Timeout::Error => e
+      print_error("Query #{domain} DNS #{type} - exception: #{e}")
+      return nil
+    end
+  end
+
+  ## TODO: Improve on-demand response (send_recv? Timeout).
+  def do_simple_get_request_raw(host, port, ssl, host_header=nil, uri, proxies)
+    begin
+      http    = Rex::Proto::Http::Client.new(host, port, {}, ssl, nil, proxies)
+      http.connect
```

Comment:
Again good use of primitives but this is implemented with nuanced considerations for the mess that is http in the exploit::http namespace

Reply:
Thank you, I have already made this remark. in fact it's my second code Ruby and I must admit that it's a bit confusing.

I did not find a use case in this context w/ scanner then I relied on modules/auxiliary/gather/censys_search.rb